### ISSUE TITLE
Incorrect Provider seen for clusters using Rancher CLI

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#37554]
**Rancher Server Setup**
- Rancher version: `rancher version v2.6.5-rc1`

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) Admin

   
**Describe the bug**
Incorrect Provider seen for clusters using Rancher CLI

**To Reproduce**
- Deploy an RKE2 node driver cluster on a docker install rancher setup
- Deploy a few RKE1 clusters also
- Login through Rancher CLI
- List clusters - RKE2 node driver cluster is seen as `Imported` and local cluster is seen as `Unknown`

**Result**
```
./rancher clusters ls

CURRENT   ID             STATE     NAME         PROVIDER                    NODES     CPU         RAM             PODS
          c-2m6sp        active    test-36455   Rancher Kubernetes Engine   6         0.33/9.50   0.34/43.39 GB   12/440
          c-b6b7s        active    test-new     Rancher Kubernetes Engine   2         0.82/8      0.34/15.38 GB   15/220
*         c-m-wcpk7gct   active    sowmya-new   Imported                    1         1.22/2      0.23/3.84 GB    18/110
          c-vkggz        active    test-15523   Rancher Kubernetes Engine   6         0.33/9.50   0.34/43.39 GB   11/440
          local          active    local        Unknown                     1         0.10/4      0.07/7.77 GB    8/110
```

**Expected:**
RKE2 node driver cluster should be `Rancher Kubernetes Engine 2` and local cluster should be `Imported`

**Comment by thedadams** [comment]
The management cluster object for RKE2 provisioned clusters is an imported cluster. So, in this case, the client would have to detect that the imported cluster is an RKE2 provisioned cluster and change the output based on that.

**Comment by jiaqiluo** [comment]
the fix can be validated in Rancher CLI `v2.6.5-rc1` https://github.com/rancher/cli/releases/tag/v2.6.6-rc1

**Comment by markusewalker** [comment]
Verified on Rancher CLI `v2.6.6-rc1`.

**ENVIRONMENT DETAILS**
- Rancher HA: `v2.6.5-rc12`
- Rancher CLI 
     - Reproduced version: `v2.6.5`
     - Verified version: `v2.6.6-rc1`
	• Token
- Browser: Chrome

**TEST RESULT**
PAASS

**REPRODUCTION STEPS**
1. Setup Rancher and navigate to the UI in a browser.
2. Created a user token for the default admin and keep note of it; will need it for later.
3. Created a standard user and login to Rancher as that user.
4. Created a user token for this standard user and keep note of it.
5. Navigated to Cluster Management and provisioned the following clusters: AKS, GKE, K3s, RKE, RKE2:
<img width="1147" alt="image" src="https://user-images.githubusercontent.com/52591819/168170098-6f330066-0286-4c72-992e-72ea4e165db0.png"> 
6. Navigated over to your client machine and make sure the Rancher CLI `v2.6.5` is installed.
7. Run command `rancher login -t <admin user token> https://<RANCHER SERVER URL>`.
8. Run command `rancher clusters ls`:

```
# rancher cluster ls
CURRENT   ID             STATE     NAME       PROVIDER                    NODES     CPU         RAM                   PODS
*         c-6q479        active    mew-gke    Unknown                     3         1.28/5.79   1325962880/16.46 GB   25/330
          c-ffjdl        active    mew-aks    Unknown                     3         1.25/5.70   0.96/13.36 GB         24/330
          c-m-gd7c4qh9   active    mew-k3s    Imported                    1         0.20/2      0.14/3.84 GB          8/110
          c-m-kmmcgnp8   active    mew-rke2   Imported                    1         1.22/2      0.23/3.84 GB          18/110
          c-mql7t        active    rketest    Rancher Kubernetes Engine   1         0.47/4      0.27/7.69 GB          10/110
```		

**VERIFICATION STEPS**
1. Setup Rancher and navigate to the UI in a browser.
2. Created a user token for the default admin and keep note of it; will need it for later.
3. Created a standard user and login to Rancher as that user.
4. Created a user token for this standard user and keep note of it.
5. Navigated to Cluster Management and provisioned/imported the following clusters: AKS, EKS, GKE, K3s, RKE, RKE2:
<img width="1257" alt="image" src="https://user-images.githubusercontent.com/52591819/168170176-8a7ba480-e7a5-4910-a067-7970b9f70371.png"> 
6. Navigated over to your client machine and make sure the Rancher CLI `v2.6.5` is installed.
7. Run command `rancher login -t <admin user token> https://<RANCHER SERVER URL>`.
8. Run command `rancher clusters ls`:

```
# rancher cluster ls
CURRENT   ID             STATE     NAME                     PROVIDER                     NODES     CPU          RAM                   PODS
          c-44cxx        waiting   mew-eks                  Elastic Kubernetes Service   0         0/0          0/0 GB                0/0
*         c-4rh5b        active    mew-aks                  Azure Kubernetes Service     1         0.94/1.90    1.50/4.45 GB          16/110
          c-m-blx8glsx   active    mew-rke2                 RKE2                         1         1.47/4       0.23/7.77 GB          13/110
          c-m-k5fsbf6x   active    mew-rke                  Imported                     1         0.47/4       0.27/7.67 GB          8/110
          c-m-rfsbqksn   active    mew-k3s                  K3S                          1         0.20/4       0.14/7.77 GB          7/110
          c-mgrwb        active    mew-gke                  Google Kubernetes Engine     3         1.28/23.73   1325962880/83.22 GB   25/330
          c-xw787        active    mew-rancher-custom-rke   Rancher Kubernetes Engine    1         0.47/4       0.27/7.67 GB          10/110
```
